### PR TITLE
feat: switch all alert queries to use lastalert

### LINKED ISSUE #2623: [➕ Feature]: Switch all Alert queries to use LastAlert

### PR DESCRIPTION
**Pull request by VladimirFilonov**
Closes #2623 
DNM before: #2473 

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

**Review by shahargl — COMMENTED** [review]
do we have some test to make sure this works? like we did with the migrations?

**Comment by Kiryous** [comment]
@coderabbitai review

**Comment by VladimirFilonov** [comment]
> do we have some test to make sure this works? like we did with the migrations?

@shahargl I did the same thing as you: compared old and new implementations. That's how I found a bug in the old one =)

```python
>>> db_alerts = get_last_alerts(tenant_id=tenant_id, limit=limit)
>>> len(db_alerts)
132
>>> len([a.fingerprint for a in db_alerts])
132
>>> len(set([a.fingerprint for a in db_alerts]))
108
```

```
>>> db_alerts_new = get_last_alerts_new(tenant_id=tenant_id, limit=limit)
>>> len(db_alerts_new)
108

>>> len(set([a.fingerprint for a in db_alerts_new]))
108

>>> set([a.fingerprint for a in db_alerts]) == set([a.fingerprint for a in db_alerts_new])
True
```

File: keep/api/models/db/migrations/versions/2024-12-02-20-42_c6e5594c99f8.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            set
+                lastalert.first_timestamp = :first_timestamp
+            where
+                lastalert.fingerprint = :fingerprint
+                and lastalert.tenant_id = :tenant_id
+        """, {
+            "first_timestamp": first_timestamp.isoformat(),
+            "fingerprint": fingerprint,
+            "tenant_id": tenant_id,
+        })
```
Comment (shahargl) [inline keep/api/models/db/migrations/versions/2024-12-02-20-42_c6e5594c99f8.py:48]: wouldn't it take lot of time to update all the table one by one? 

maybe smth like?
```
UPDATE lastalert
SET first_timestamp = (
    SELECT MIN(alert.timestamp)
    FROM alert
    WHERE alert.fingerprint = lastalert.fingerprint 
    AND alert.tenant_id = lastalert.tenant_id
)
```

**Review by shahargl — APPROVED** [review]
lgtm

**Pull request merged**